### PR TITLE
cylc refresh: don't nuke passphrase on title change

### PR DESCRIPTION
**Pull request by matthewrmshin**
It was nuking the passphrase by doing an unregister and a new register
on suite title change. This is not necessary, and on running suites,
will cause subsequent client commands to fail.

Also modified `cylc.registration` to improve its pylint score.

The new test would time out before this change, because the refresh
command would reset the passphrase, which would make it impossible for
subsequent task jobs to report progress back to the suite.

Related to metomi/rose#1846. @hjoliver @arjclark please review.

### CODE REVIEW

File: lib/cylc/registration.py [hunk 1]
```diff
@@ -15,53 +15,55 @@
...
             suites = os.listdir(self.dbpath)
-        except Exception, x:
-            sys.exit(str(x))
+        except OSError as exc:
+            sys.exit(str(exc))
         return suites
 
-    def register(self, name, dir):
+    def register(self, name, path):
+        """Register a suite, its source patha nd its title."""
```
Comment (hjoliver) [inline lib/cylc/registration.py]: "patha nd"

File: tests/cylc-refresh/00-title.t [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+# Test "cylc refresh" on a running suite, with a title change to "suite.rc".
```
Comment (hjoliver) [inline tests/cylc-refresh/00-title.t]: Can you a few words to this comment to explain to our future selves why "cylc refresh" needs to be tested in this way (or just reference the github issue).

**Comment by hjoliver** [comment]
Good, tests all pass. Over to Andy.

**Comment by arjclark** [comment]
Review 2 - looks ok to me. Once @hjoliver's comments are addressed is good for merging.

File: lib/cylc/registration.py (hunk 1 above)
Comment (matthewrmshin) [inline lib/cylc/registration.py]: Fixed.

File: tests/cylc-refresh/00-title.t (hunk 2 above)
Comment (matthewrmshin) [inline tests/cylc-refresh/00-title.t]: Comment added.

**Comment by matthewrmshin** [comment]
@arjclark all problems addressed.

**Pull request merged**